xcube_cds/store.py

```diff
@@ -749,11 +764,26 @@ def open_data(self, data_id: str, opener_id: Optional[str] = None,
     # Implementation helpers
 
     @staticmethod
-    def _assert_valid_type_id(type_id):
-        if type_id is not None and type_id != TYPE_ID_DATASET:
+    def _validate_type_specifier(type_specifier: Union[str, TypeSpecifier]):
+        if not CDSDataStore._is_type_specifier_compatible(type_specifier):
             raise DataStoreError(
-                f'Data type identifier must be "{TYPE_ID_DATASET}", '
-                f'but got "{type_id}"')
+                f'Supplied type specifier "{type_specifier}" is not compatible '
+                f'with "{TYPE_SPECIFIER_CUBE}."'
+            )
+
+    @staticmethod
+    def _is_type_specifier_compatible(
```

Comment:
Even though this is just an internal method, I think it would be better to change its name to satisfies_type_specifier (or similar)

Reply:
Agreed. I've changed it to `_is_type_specifier_satisfied`.